.github/workflows/run-tests.yml

```diff
@@ -77,6 +77,68 @@ jobs:
           name: code-coverage
           path: coverage/.coverage.py*
 
+  min-deps-test:
+    # Run the TOX-ENV tests for the given OS and python VERSION with
+    # the minimal possible Basilisp dependencies versions.
+    #
+    # For dependencies to be elligible for minimum version testing,
+    # they should be declared in `pyproject.toml` as such in the
+    # following format
+    #
+    # <dependency-name> = "^..."
+    # <dependency-name> = ">=..."
+    # <dependency-name> = { version = "^..." ...
+    # <dependency-name> = { version = ">=..." ...
+    #
+    # Note that the intention is to modify only the lines in
+    # [tool.poetry.dependencies]. However, the current implementation
+    # for simplicity will blindly alter all lines starting with the
+    # above patterns in the file. While this includes
+    # [tool.poetry.dev-dependencies] too (which is deemed acceptable),
+    # it is important to consistently assess that no other section is
+    # adversely affected in the future.
```

Comment:
I think the commentary is not necessary. As I said in another comment, all dependencies always resolve together. If for some reason one group of dependencies become incompatible with another, that would be an error worthy of fixing in its own right (and would likely cause Poetry to fail on other jobs as well).

As to the possibility that this alters the file in unexpected ways, we also have the option to modify the `pyproject.toml` more holistically in a prior step using a TOML library. Still, taking a look at the file I don't see any lines of particular concern.

Ultimately, because of the open ended nature of defining dependency constraints and us not having the tools at our disposal to manipulate them properly, this step is never going to be perfect. The best we can hope for is that it catches some problematic cases before users have to report them.

Reply:
Removed.